### DISCUSSION TITLE
in-sub editor text pasting of item assemblies

### DISCUSSION THREAD

**Discussion by Tsunder** [FakeFishGames/Barotrauma#4912] [Ideas and suggestions]
A way to allow copy/pasting of text assemblies by text would be hugely nice QoL improvement for sub editors (perhaps compressed)

currently process is:
create item assembly xml
open folder
upload file

other person downloads file
moves file over to item assemblies folder
restarts game/ starts game
open sub editor
open sub
finally, paste sub assembly

in-editor pasting would make it exactly 94.2x more convenient with:
create item assembly xml (copied to clipboard)
share text

other person copies text to clipboard
(starts game if necessary but this can happen while sender was preparing assembly)
opens "import/export item assembly"
pastes clipboard
item assembly made and selected!

**Comment by 3e849f2e5c** [comment]
I'm a huge fan of this idea, I'll look into it when I have time

**Comment by mygamingaccount** [comment]
XML assemblies can get huge, some networks might rate-limit the transmissions!

Please consider compressing the XML with GZIP first, then encoding the result using base64!
(relevant info: Factorio Blueprints are gzip compressed JSON, encoded in base64)
